src/test/java/ibm/jceplus/junit/base/BaseTestECDHKeyAgreement.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright IBM Corp. 2024
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution.
+ */
+package ibm.jceplus.junit.base;
+
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.interfaces.ECPrivateKey;
+import java.security.spec.ECPrivateKeySpec;
+import javax.crypto.KeyAgreement;
+import org.junit.jupiter.api.Test;
+
+public class BaseTestECDHKeyAgreement extends BaseTestJunit5 {
+
+    @Test
+    public void testInitWithInvalidKey() throws Exception {
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC", getProviderName());
+        kpg.initialize(256);
+        KeyPair kp = kpg.generateKeyPair();
+        ECPrivateKey privateKey = (ECPrivateKey) kp.getPrivate();
+
+        KeyFactory keyFactory = KeyFactory.getInstance("EC", getProviderName());
+        ECPrivateKey invalidPrivateKey
+                = (ECPrivateKey) keyFactory.generatePrivate(
+                        new ECPrivateKeySpec(BigInteger.ZERO,
+                                privateKey.getParams()));
+
+        KeyAgreement ka = KeyAgreement.getInstance("ECDH", getProviderName());
+
+        // The first initiation should succeed.
+        ka.init(privateKey);
+
+        // The second initiation should fail with invalid private key,
+        // and the private key assigned by the first initiation should be cleared.
+        try {
+            ka.init(invalidPrivateKey);
+        } catch (java.security.InvalidKeyException ike) {
+            System.out.println("Expected <java.security.InvalidKeyException> is caught.");
+        }
+
+        // Cannot doPhase due to no private key.
+        try {
+            ka.doPhase(kp.getPublic(), true);
```

Comment:
Should we have the following lines here given that the test is called `testInitWithInvalidKey`? We already test `doPhase()` in the next method and maybe we should have a separate method to check the `generateSecret()`.

Reply:
I removed `doPhase` and `generateSecret` tests since modifications are only applied to the `engineInit()` in `ECDHKeyAgreement` and `engineInitSign` in `ECDSASignature`. Did not change the doPhase and generateSecret actions.